cmd/list.go

```diff
@@ -44,13 +45,90 @@ const (
 
 	cellSideCount = 2
 
+	headerCount = 4
+
 	paddingHorizontal = 1
 	paddingVertical   = 0
 
 	rowHeaderIdx = 0
 	rowMaxHeight = 1
 )
 
+func getTermInfo() int {
+	width, _, err := term.GetSize(0)
+	if err != nil {
+		return 0
+	}
+
+	return width
+}
+
+func determineColumnWidths(rows ...[]string) (tooWide bool, colWidths []int) {
+	terminalWidth := getTermInfo()
+	idColumnWidth := columnWidthID
+	aliasColumnWidth := columnWidthAlias
+
+	for _, row := range rows {
+		if idWidth := len(row[columnIdxID]); idWidth >= idColumnWidth {
+			idColumnWidth = idWidth + (paddingHorizontal * cellSideCount)
+		}
+
+		if aliasWidth := len(row[columnIdxAlias]); aliasWidth >= aliasColumnWidth {
+			aliasColumnWidth = aliasWidth + (paddingHorizontal * cellSideCount)
+		}
+	}
+
+	totalWidth := (idColumnWidth + aliasColumnWidth + columnWidthVersion + columnWidthNumNodes)
+	totalWidth += headerCount * (paddingHorizontal * cellSideCount)
+
+	return (totalWidth > terminalWidth), []int{idColumnWidth, aliasColumnWidth, columnWidthVersion, columnWidthNumNodes}
+}
+
+func printList(rows ...[]string) {
+	for _, row := range rows {
+		fmt.Fprintf(os.Stdout, "ID: %s\n"+"Alias: %s\n"+"Version: %s\n"+"# Nodes: %s\n\n",
+			row[columnIdxID], row[columnIdxAlias], row[columnIdxVersion], row[columnIdxNumNodes])
+	}
+}
```

Comment:
I don't think the print should happen in the function directly, it should just return a formatted string to print from the calling context

```suggestion
func formatList(rows ...[]string) string {
	output := ""

	for _, row := range rows {
		output = strings.Join([]string{
			output,
			fmt.Sprintf("ID      : %s", row[columnIdxID]),
			fmt.Sprintf("Alias   : %s", row[columnIdxAlias]),
			fmt.Sprintf("Version : %s", row[columnIdxVersion]),
			fmt.Sprintf("# Nodes : %s", row[columnIdxNumNodes]),
			"",
		}, "\n")
	}

	return output
}
```

and call it with `fmt.Fprint(os.Stdout, formatList(rows...))`